### ISSUE TITLE
False negative on template v-for with multiple children

### ISSUE THREAD

**Issue by trygveaa** [vuejs/eslint-plugin-vue#1412]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** v7.17.0
- **eslint-plugin-vue version:** 7.4.1
- **Node version:** v15.5.1
- **Operating System:** Arch Linux

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
  },
  extends: [
    '@nuxtjs/eslint-config-typescript',
    '@nuxtjs',
    'plugin:nuxt/recommended',
    'plugin:prettier/recommended',
    'plugin:vue/vue3-recommended',
    'prettier',
    'prettier/vue',
  ],
  plugins: ['prettier'],
  // add your custom rules here
  rules: {
    'no-unused-expressions': 'off',
    '@typescript-eslint/no-unused-expressions': ['error'],

    // Disable until we are using Vue 3
    'vue/no-deprecated-v-on-native-modifier': 'off',
  },
}
```

**What did you do?**	
```vue
<template>
  <div>
    <template v-for="x in list">
      <div :key="x.key">1</div>
      <div v-for="y in x.sublist" :key="y.key">2</div>
    </template>
  </div>
</template>
```

**What did you expect to happen?**
I expected to get no errors.

**What actually happened?**
```
4:12  error  `<template v-for>` key should be placed on the `<template>` tag   vue/no-v-for-template-key-on-child
```

If I try to move the first key to the template tag I get this error instead:
```
3:33  error  '<template v-for>' cannot be keyed. Place the key on real elements instead  vue/no-v-for-template-key
```

Additionally, if I do this, Vue errors, while if I keep the keys as in the code above Vue doesn't complain and renders correctly, only eslint complains.

As far as I understand, when you have multiple children in a template tag, the keys have to be placed on the children, not on the template, so therefore the rule should not give any error.

**Repository to reproduce this issue**
https://github.com/trygveaa/eslint-plugin-vue/commit/31702ced9fb0f74ee668dd1d26452d9820cdb5c3

**Comment by ota-meshi** [comment]
The `vue/no-v-for-template-key` rule is for Vue 2. If you use Vue 3 you should not use it rule.
The preset `plugin:vue/vue3-recommended` does not contain the `vue/no-v-for-template-key` rule, so other presets may contain it rule.
A simple workaround is to add `"vue/no-v-for-template-key": "off"`.

**Comment by trygveaa** [comment]
Ah, okay, thanks. As mentioned, if we follow `vue/no-v-for-template-key-on-child` we get errors from Vue. I guess this is because we're not using Vue 3 yet, just `@vue/composition-api`. I'll disable `vue/no-v-for-template-key-on-child` for now, and check it out again when we upgrade to Vue 3 proper.
